go/stellar/transform.go

```diff
@@ -212,6 +212,16 @@ func transformPaymentRelay(mctx libkb.MetaContext, acctID stellar1.AccountID, p
 		loc.ToType = stellar1.ParticipantType_STELLAR
 		loc.ToUsername = ""
 		loc.ToAccountName = ""
+		if loc.StatusSimplified == stellar1.PaymentStatus_CANCELED {
+			// canceled payment. blank out toAssertion and stow in originalToAssertion
+			// set delta to what it would have been had the payment completed
+			loc.ToAssertion = ""
+			loc.OriginalToAssertion = p.ToAssertion
+			loc.Delta = stellar1.BalanceDelta_INCREASE
+			if acctID == p.FromStellar {
+				loc.Delta = stellar1.BalanceDelta_DECREASE
+			}
+		}
```

Comment:
Why would you want delta to be what could have but definitely won't happen? If you canceled a relay back into the same account intuitively the delta is NONE as net no money has moved.

Btw I think it's possible to send a relay from primary A, change your primary to B, and cancel it. And it will have gone from A to B. Just a random thought.

Reply:
The views for `status == 'claimable'` and `status == 'canceled'` are very similar. We use the delta to figure out some signs and icons (+/-, sending/receiving), which should be the same in the two cases.

I'm not sure your example will work like that - [`CancelPaymentLocal`](https://github.com/keybase/client/blob/master/go/stellar/stellarsvc/frontend.go#L484) uses `relay.FromStellar` for `into`.

Reply:
So if a user deletes an account that they sent an unclaimed relay from then they lose the money even though technically there's a box for them with the secret key?

Reply:
Well, if they're claiming the payment from the UI then they haven't deleted the account. They'd have to use the CLI which falls back to the default account ID

Reply:
I mean if someone sends and then deletes the sending account. There's no way for them to reclaim the money. Seems unfortunate. Sort of makes sense.

Reply:
yes